### PR TITLE
Fix FixedRingBuffer.removeAt

### PR DESCRIPTION
**Pull request by tchaloupka**
There is a problem when removing from the start index

### CODE REVIEW

File: source/vibe/utils/array.d [hunk 1]
```diff
@@ -376,7 +376,7 @@ struct FixedRingBuffer(T, size_t N = 0, bool INITIALIZE = true) {
 		assert(r.m_buffer is m_buffer);
 		if( m_start + m_fill > m_buffer.length ){
 			assert(r.m_start >= m_start && r.m_start < m_buffer.length || r.m_start < mod(m_start+m_fill));
-			if( r.m_start > m_start ){
+			if( r.m_start >= m_start ){
```
Comment (s-ludwig) [inline source/vibe/utils/array.d]: Would probably be a nice optimization to do `popFront` in case of `r.m_start == m_start`.

**Comment by s-ludwig** [comment]
Sounds like this might have been responsible for some obscure bugs!

File: source/vibe/utils/array.d (hunk 1 above)
Comment (tchaloupka) [inline source/vibe/utils/array.d]: Good point, fixed

Comment (s-ludwig) [inline source/vibe/utils/array.d]: Thanks! Will merge when Travis is done.

**Comment by tchaloupka** [comment]
I just squashed the PR.

**Comment by s-ludwig** [comment]
Just discovered that GitHub now also offers this as an option for how to merge the PR, which is pretty nice for avoiding to bother the PR authors with this!

**Pull request merged**